sdk/lib/socks5-listener/src/lib.rs

```diff
@@ -74,19 +78,38 @@ fn rust_free_string(string: char_p::Box) {
 #[derive_ReprC]
 #[ffi_export]
 #[repr(u8)]
+#[derive(Eq, PartialEq)]
 pub enum ClientState {
-    Unknown,
+    Uninitialised,
     Connected,
     Disconnected,
 }
 
+#[ffi_export]
+pub fn get_client_state() -> ClientState {
+    // if the environment is not set, we never called start before
+    // if the shutdown was never set, the client can't possibly be running
+    // and similarly if it's set, it's most likely running
+    if !ENV_SET.load(Ordering::Relaxed) {
+        ClientState::Uninitialised
+    } else if RUNTIME.block_on(is_shutdown_handle_set()) {
+        ClientState::Connected
+    } else {
+        ClientState::Disconnected
+    }
+}
+
 #[ffi_export]
 pub fn start_client(
     storage_directory: Option<char_p::Ref<'_>>,
     service_provider: Option<char_p::Ref<'_>>,
     on_start_callback: StartupCallback<'static>,
     on_shutdown_callback: ShutdownCallback<'static>,
 ) {
+    if get_client_state() == ClientState::Connected {
+        return;
```

Comment:
Adding a log could be useful to track useless calls to `start_client`/`stop_client`